From 3c82cfbddb1143e6fe64438f380e61e45c2bfe0f Mon Sep 17 00:00:00 2001
From: Aiswarya Cyriac <aiswarya.cyriac@intel.com>
Date: Tue, 4 Jun 2019 09:29:39 +0530
Subject: [PATCH] Fix Possible null pointer dereference in btusb_sco_snd_card
 driver

dereference of dynamic memory area which is freed using kfree is
fixed

Tracked-On: OAM-81032
Signed-off-by: Aiswarya Cyriac <aiswarya.cyriac@intel.com>
---
 sound/usb/btusb/btusb_sco_snd_card.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/sound/usb/btusb/btusb_sco_snd_card.c b/sound/usb/btusb/btusb_sco_snd_card.c
index b2cbb9ca8f42..a043aefc5ef1 100644
--- a/sound/usb/btusb/btusb_sco_snd_card.c
+++ b/sound/usb/btusb/btusb_sco_snd_card.c
@@ -127,8 +127,7 @@ static void deinit_capture_data_cb(struct capture_data_cb *cb)
 {
 	kfree(cb->buf);
 	kfree(cb);
-	cb->pos = 0;
-	cb->expected = 0;
+	cb = NULL;
 }
 
 static inline void __fill_isoc_descriptor(struct urb *urb, int len, int mtu)
-- 
2.17.1

